sdk/textanalytics/Azure.AI.TextAnalytics/src/AuthenticationPolicy.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.AI.TextAnalytics
+{
+    internal class AuthenticationPolicy : HttpPipelineSynchronousPolicy
+    {
+        private const string AuthorizationHeader = "Ocp-Apim-Subscription-Key";
+        private TextAnalyticsSubscriptionKeyCredential _credential;
+
+        public AuthenticationPolicy(TextAnalyticsSubscriptionKeyCredential credential)
+        {
+            _credential = credential;
+        }
+
+        public override void OnSendingRequest(HttpMessage message)
+        {
+            message.Request.Headers.Add(AuthorizationHeader, _credential.SubscriptionKey);
```

Comment:
Are there any thread safety considerations here?  i.e. if someone was updating the key at the same time as this request was being sent?

Reply:
Because there is only one field here I don't believe you have any thread safety issues. The subscription key could be read before or after the update, but it will never be read in an inconsistent or torn state, which is really we need to guard against in the library. If synchronizing client calls with the update of the subscription key is necessary I would say that this is the callers responsibility.

Reply:
Sounds good.  Thanks @schaabs !!